src/de/mod.rs

```diff
@@ -77,12 +77,7 @@ pub enum Error {
     CustomErrorWithMessage(heapless::String<64>),
 }
 
-#[cfg(feature = "std")]
-impl ::std::error::Error for Error {
-    fn description(&self) -> &str {
-        ""
-    }
-}
+impl serde::de::StdError for Error {}
```

Comment:
I think this still needs a `std` feature for the impl. This pulls in `std::error::Error` as a re-export in serde: https://docs.rs/serde/1.0.147/serde/de/index.html#reexports

Reply:
It correctly muxes for non-std:
https://docs.rs/serde/1.0.147/src/serde/de/mod.rs.html#134

Reply:
It indeed does! Thanks for clarifying his. Maybe worth a comment in the code that serde handles muxing of it automagically?

Reply:
Not really necessary IMO. The code would be completely broken otherwise, fail to build everywhere, and would also fail CI. And we are just doing what the [serde docs say](https://github.com/serde-rs/serde/blob/master/serde/src/std_error.rs).